examples/dielectric.jl

```diff
@@ -0,0 +1,54 @@
+# Compute a few eigenvalues of the dielectric matrix (q=0,ω=0) iteratively
+
+using DFTK
+using Plots
+using KrylovKit
+
+# Calculation parameters
+kgrid = [1, 1, 1]
+Ecut = 5
+
+# Silicon lattice
+a = 10.26
+lattice = a / 2 .* [[0 1 1.]; [1 0 1.]; [1 1 0.]]
+Si = ElementPsp(:Si, psp=load_psp("hgh/lda/Si-q4"))
+atoms = [Si => [ones(3)/8, -ones(3)/8]]
+
+# change the symmetry to compute the dielectric operator with and without symmetries
+model = model_LDA(lattice, atoms, symmetry=:off)
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid)
+scfres = self_consistent_field(basis, tol=1e-14)
+
+# Apply ε = 1 - K χ0
+function epsfun(dv)
+    dv = reshape(dv, size(scfres.ρ.real))
+    dρ = apply_χ0(scfres.ham, dv, scfres.ψ, scfres.εF, scfres.eigenvalues)
+    Kdρ = apply_kernel(basis, dρ; ρ=scfres.ρ)
+    vec(dv - Kdρ)
+end
+
+# A straightfoward Arnoldi eigensolver that diagonalizes the matrix at each step
+# This is more efficient than Arpack when `f` is very expensive
+function arnoldi(f, x0, howmany; tol=1e-4, maxiter=30)
+    for (V, B, r, nr, b) in ArnoldiIterator(f, x0)
+        # A * V = V * B + r * b'.
+        V = hcat(V...)
+        AV = V*B + r*b'
+
+        ew, ev = eigen(B, sortby=real)
+        Vr = V*ev
+        AVr = AV*ev
+        R = AVr - Vr * Diagonal(ew)
```

Comment:
Can you indicate by a comment that this is the residual (or call it residual) ... just to make it clear.

Reply:
if I do that I should probably comment every line... this is kind of a hack right now, it should be turned into a proper eigensolver eventually, but I'm too lazy to do that, so this is an exploratory tool for now

Reply:
ok.